packages/webpack-plugin/src/calc-depth.ts

```diff
@@ -0,0 +1,80 @@
+import { join, parse } from 'path';
+import { Module, ModuleGraph, NormalModule } from 'webpack';
+import { isStylableModule, uniqueFilterMap } from './plugin-utils';
+
+export interface DepthResults {
+    id: string;
+    depth: number;
+    deps: Set<Module>;
+}
+
+export function calcDepth(
+    module: Module,
+    moduleGraph: ModuleGraph,
+    path: Module[] = [],
+    cache = new Map<Module, number>()
+): number {
+    let cssDepth = 0;
+
+    if (cache.has(module)) {
+        return cache.get(module)!;
+    }
+
+    if (path.includes(module)) {
```

Comment:
rather than tracking circularity using arrays of Modules, can we replace this if with: `cache.set(module, 0)`?